File: backend/src/main/java/com/readnest/service/RecordService.java
package com.readnest.service;

import com.readnest.pojo.Detail;
import com.readnest.pojo.Pages;
import com.readnest.pojo.Record;
import com.readnest.pojo.userRecord;

import java.util.Date;
import java.util.List;

public interface RecordService {
    Pages fetchRecords(Integer page, Integer pageSize);

    Pages searchRecords(Integer page, Integer pageSize, String status, String searchItem, Date startDate, Date endDate);

    Detail getRecordById(String recordId);

    List<userRecord> getRecordsByUsername(String username);
}

File: Frontend/src/service/recordService.ts
import { http } from "@/utils";

export const fetchRecords = async (Page: number, PageSize: number,) => {
    return await http.get("/api/recordsData", {
        params: {
            page: Page,
            pageSize: PageSize
        }
    });
};

//when searching records
export const searchRecords = async (Page: number, PageSize: number, SearchItem: string, startDate: string, endDate: string, status: string) => {
    return await http.get("/api/searchRecords", {
        params: {
            page: Page,
            pageSize: PageSize,
            status: status,
            SearchItem: SearchItem,
            startDate: startDate,
            endDate: endDate
        }
    });
}

export const fetchDetail = async (recordId: string | undefined) => {
    return await http.get("/api/fetchRecord", {
        params: {
            recordId: recordId
        }
    });
};